### PR TITLE
react-dnd definition: Add DragDropContextProvider

### PR DESCRIPTION
**Pull request by Stephen2**
A duplicate of this: https://github.com/flow-typed/flow-typed/pull/1666 but rebased

### CODE REVIEW

**Comment by Stephen2** [comment]
Sorry, not familiar with the CI process, are these the cause of the failure:
```
   Warning: test_react-dnd-v2.js:413
   413: // $ExpectError: Invalid default prop
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Error suppressing comment. Unused suppression
```

**Comment by Stephen2** [comment]
Sorry for the mess, should be all sorted now

File: definitions/npm/react-dnd_v2.x.x/flow_v0.53.x-/test_react-dnd-v2.js [hunk 1]
```diff
@@ -288,7 +285,6 @@ const dndKnight3: DndKnight3 = (null: any);
   typeof DndKnight3
 >);
 
-// $ExpectError: Invalid collect prop
```
Comment (villesau) [inline definitions/npm/react-dnd_v2.x.x/flow_v0.53.x-/test_react-dnd-v2.js]: @Stephen2 why these are removed?

Comment (Stephen2) [reply definitions/npm/react-dnd_v2.x.x/flow_v0.53.x-/test_react-dnd-v2.js]: Tests failed.  I suppose these things no longer fail, so $ExpectError was wrong now?  I don't really understand the testing, but removing these fixed failing tests

**Pull request merged**